direct-messages/README.md

```diff
@@ -26,41 +35,48 @@ const hash = 'SHA256'
 const length = 32
 const salt = SHA256('envelope-dm-v1-extract-salt')
 
-function directMessageKey (my_dh_secret, my_dh_public, my_feed_tfk, your_dh_public, your_feed_tfk) {
-  var  input_keying_material = scalarmult(my_dh_secret, your_dh_public)
+function directMessageSlotKey (my_dh_secret, my_dh_public, my_id_bfe, your_dh_public, your_id_bfe) {
+  var input_keying_material = scalarmult(my_dh_secret, your_dh_public)
 
   var info_context = Buffer.from('envelope-ssb-dm-v1/key', 'utf8')
   var info_keys = sort([
-    my_dh_public || my_feed_tfk,
-    your_dh_public || your_feed_tfk
```

Comment:
I noticed that in `ssb-private-group-keys` we were not following this spec!
Instead we were using the BFE/TFK encoding of the diffie-hellman keys on each line.

However the test vectors followed the clarified definition on the right, so if you passed `vectores/direct-message-key1.json` then you should be all good.

cc @cryptix @arj03

Reply:
Here's those same lines in the js implementation : https://github.com/ssbc/ssb-private-group-keys/blob/master/direct-message-key.js#L21-L22

The gotcha was that the function signature in JS takes ALL BFE buffers, whereas the algorithm in this README uses a mix of some BFE and some raw buffers....

Reply:
Thanks for the shout. Just checked and go-ssb only used the envelope spec until now.

Reply:
The envelope spec doesn't let you do anything in scuttlebutt by itself...
do you mean you haven't implemented any part of the private groups spec? (this is the dm part of that spec, but the group key part defines what a group ID is as well and how to map that to a key)